server/src/main/java/io/crate/exceptions/UnsupportedFunctionException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.exceptions;
+
+public class UnsupportedFunctionException extends RuntimeException implements ResourceUnknownException, SchemaScopeException {
+
```

Comment:
another thought on this: what if we turn `UnsupportedFeatureException` into `UnscopedException`? 
This way we avoid creation of a new `Exception` class and avoid fixing multiple tests.

The only controversial thing is whether we need to check scope if users try to do something in some forbidden for them scope as making it `UnscopedException` basically skips all checks.

To me `UnsupportedFeatureException` looks like special case where we can tell users right away that it's not supported regardless of their permissions - even if user requests a missing permission, error will still pop up - so maybe just cut it short and skip the checks?

Reply:
Currently we execute statements according to the privileges through `ensureMayExecute` and any exceptions thrown are translated by `ensureMaySee`. I think your suggestion not only apply to `UnsupportedFeatureException` but other execution paths below `ensureMaySee`.
Also, current behaviour aligns with Postgres. Maybe this can be an enhancement out side of this bug fix?

Reply:
I thought ensureMaySee already skips scope checks for multiple paths (all implementations of `UnscopedException`) but I'm fine with the current approach as well.